configure

```diff
@@ -196,6 +195,11 @@ if [ -n "$NETCDFPAR" ] ; then
   export NETCDF
   export NETCDF4
   export USENETCDFPAR
+else
+    # Makefile assumes USENETCDFPAR is set
+    # Line 1072
+    USENETCDFPAR="0"
+    export USENETCDFPAR
```

Comment:
While I agree that the additions here in an `else` block make sense, it turns out that they can actually break configuration in some cases. I've got a NetCDF library that was build to support parallel I/O and that requires linking with the MPI library, but I don't set the `NETCDFPAR` environment variable. By exporting `USENETCDFPAR=0`, the code around line 1072 in the Makefile, i.e.,
```
        if [ $(USENETCDFPAR) -eq 0 ] ; then \
         ( cd tools ; /bin/rm -f nc4_test.{exe,nc,o} ; $(SCC) -o nc4_test.exe nc4_test.c -I$(NETCDF)/include -L$(NETCDF)/lib -lnetcdf $(NETCDF4_DEP_LIB) ; cd .. ) ; \
        else \
         ( cd tools ; /bin/rm -f nc4_test.{exe,nc,o} ; $(DM_CC) -o nc4_test.exe nc4_test.c -I$(NETCDF)/include -L$(NETCDF)/lib -lnetcdf $(NETCDF4_DEP_LIB) ; cd ..  ) ; \
        fi
```
then tries to build a test program using the serial Fortran compiler rather than the MPI Fortran wrapper, resulting in a failure.

Ultimately, I think this is our problem, and we should consider making our configuration checks more robust. For example, the code above from the main `Makefile` could be modified as follows:
```patch
 # rule used by configure to test if this will compile with netcdf4
 nc4_test:
-	if [ $(USENETCDFPAR) -eq 0 ] ; then \
-	 ( cd tools ; /bin/rm -f nc4_test.{exe,nc,o} ; $(SCC) -o nc4_test.exe nc4_test.c -I$(NETCDF)/include -L$(NETCDF)/lib -lnetcdf $(NETCDF4_DEP_LIB) ; cd .. ) ; \
-	else \
-	 ( cd tools ; /bin/rm -f nc4_test.{exe,nc,o} ; $(DM_CC) -o nc4_test.exe nc4_test.c -I$(NETCDF)/include -L$(NETCDF)/lib -lnetcdf $(NETCDF4_DEP_LIB) ; cd ..  ) ; \
-	fi
+	 ( cd tools ; /bin/rm -f nc4_test.{exe,nc,o} ; $(CC) -o nc4_test.exe nc4_test.c -I$(NETCDF)/include -L$(NETCDF)/lib -lnetcdf $(NETCDF4_DEP_LIB) ; cd .. )
 
 # rule used by configure to test if Fortran 2003 IEEE signaling is available
 fortran_2003_ieee_test:
```
Let's see if we can fix up WRF's configuration system before the v4.5 release.

Reply:
Are you leaning toward #1743 to fix this particular issue then?  And perhaps update the documentation as to how `USENETCDFPAR` is supposed to be used, since your understanding evidently doesn't match mine?

Reply:
> While I agree that the additions here in an `else` block make sense, it turns out that they can actually break configuration in some cases. I've got a NetCDF library that was build to support parallel I/O and that requires linking with the MPI library, but I don't set the `NETCDFPAR` environment variable. 

From [the README for parallel NetCDF using NetCDF4](https://github.com/wrf-model/WRF/blob/master/doc/README.netcdf4par), it looks like you should be setting `NETCDFPAR`:
https://github.com/wrf-model/WRF/blob/21c72141142fc6c8d203d2bf79f1990e45a0aef8/doc/README.netcdf4par#L7-L13

Should this go in the install documentation somewhere?  Where would I change that?

Reply:
The reason I decided to include this change is that I set up an environment with everything needed for a serial compilation of WRF with netCDF support but without MPI, which failed because this check thought it had to use `mpicc` to link against my (serial) netCDF library.  The options were:
1. Fix the check so it wouldn't try to use `mpicc`
2. Dig through the documentation for an option I'm not using that isn't available on my platform to determine the way I would request that behavior and use that to back-engineer how I would disable that option.

I feel the first option is more inline with the existing documentation in `doc/README.netcdf4par`, and that the file is reasonably named and located for anyone who wants to re-enable that optional behavior.  I leave it up to the developers to determine how to fix this check, whether [the option you proposed above](https://github.com/wrf-model/WRF/pull/1812#discussion_r1139462243), the option I propose here, the option proposed in #1743, or something else.

Reply:
I removed this code, so WRF configure again assumes that netCDF4==MPI unless the user knows to set `USENETCDFPAR=0`.  I also documented this requirement in `doc/README.netcdf4par`.